grizli/prep.py

```diff
@@ -3179,11 +3219,19 @@ def process_direct_grism_visit(direct={},
                 changed_filter = False
 
             # Run updatewcs
-            try:
-                updatewcs.updatewcs(file, verbose=False, use_db=False)
-            except:
-                updatewcs.updatewcs(file, verbose=False)
-
+            if isJWST:
+                img = jwst_utils.img_with_wcs(file)
+                img.save(file)
+                temp_hdu = pyfits.open(file,mode='update')
+                temp_hdu[0].header['FILTER'] = temp_hdu[0].header['FILTER_INFO']
+                temp_hdu[0].header['EXP_TYPE'] = temp_hdu[0].header['EXPTYPE_INFO']
+                temp_hdu.flush()
+            else:
+                try:
+                    updatewcs.updatewcs(file, verbose=False, use_db=False)
+                except:
```

Comment:
Same as above for `except TypeError`.  Not great to have things duplicated from the direct image processing, but the handling does need to be different here.  Also will have to update here if you change the placeholder header keywords like I mentioned above.

Reply:
Done in the same commit as above: 22b1a7847907cfedec10eb55d3566a21fb9e3b4d